### ISSUE TITLE
enforce-canonical is too limited to negative

### ISSUE THREAD

**Issue by tleunen** [sergioazoc/oxlint-tailwindcss#11]
[TailwindCSS intellisense](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) in vscode has a way to suggest canonical classes (`suggestCanonicalClasses`) and does report a lot more than what the current `enforce-canonical` rule is reporting.

To give an example, the vscode extension reports `p-[2px]` to be `p-0.5` instead but the lint rule does not. I don't think that one fits the `no-arbitrary-values` because in some cases, you want to allow arbitrary values, just not allow them when they have canonical values instead?

But here are a few examples of the extension to give a few ideas
````
The class `max-w-[400px]` can be written as `max-w-100`(suggestCanonicalClasses)
The class `[--w-padding:theme(spacing.1)]` can be written as `[--w-padding:--spacing(1)]`(suggestCanonicalClasses)
The class `text-[var(--color-text)]/90` can be written as `text-(--color-text)/90`(suggestCanonicalClasses)
```

**Comment by sergioazoc** [comment]
Great suggestion @tleunen — fixed in v0.6.0.

`enforce-canonical` now uses the exact same API as TailwindCSS IntelliSense: `ds.canonicalizeCandidates(classes, { rem })` called dynamically via a persistent worker thread. All your examples are now detected:

```
p-[2px]                           → p-0.5
max-w-[400px]                     → max-w-100
text-[var(--color-text)]/90       → text-(--color-text)/90
[--w-padding:theme(spacing.1)]    → [--w-padding:--spacing(1)]
```

The px→named conversion (like `p-[2px]` → `p-0.5`) depends on `rootFontSize` which defaults to 16 (same as the IntelliSense extension). Configurable via:

```jsonc
{
  "settings": {
    "tailwindcss": {
      "rootFontSize": 16
    }
  }
}
```

The precomputed cache is kept as fallback if the worker isn't available, so existing behavior is preserved.
